idaes/surrogate/pysmo_surrogate.py

```diff
@@ -0,0 +1,759 @@
+        ),
+    )
+
+    CONFIG.declare(
+        "number_of_crossvalidations",
+        ConfigValue(
+            default=3, domain=PositiveInt, description="Number of crossvalidations."
+        ),
+    )
+
+    CONFIG.declare(
+        "training_split",
+        ConfigValue(
+            default=0.8,
+            domain=PositiveFloat,
+            description="Training-testing data split for PySMO.",
+        ),
+    )
+
+    CONFIG.declare(
+        "solution_method",
+        ConfigValue(
+            default=None,
+            domain=In(["pyomo", "mle", "bfgs"]),
+            description="Method for solving regression problem. Must be one of the options ['pyomo', 'mle', 'bfgs']. ",
+        ),
+    )
+
+    CONFIG.declare(
+        "multinomials",
+        ConfigValue(
+            default=False,
+            domain=Bool,
+            description="Option for bi-variate pairwise terms in final polynomial",
+        ),
+    )
+
+    CONFIG.declare(
+        "extra_features",
+        ConfigValue(
+            default=None,
+            domain=list,
+            description="List of extra features to be considered for regression (if any), e.g. ['x1 / x2']. ",
+        ),
+    )
+
+
+    def __init__(self, **settings):
+        super().__init__(**settings)
+
+    def _create_model(self, pysmo_input, output_label):
+        model = pr.PolynomialRegression(
+            pysmo_input,
+            pysmo_input,
+            maximum_polynomial_order=self.config.maximum_polynomial_order,
+            training_split=self.config.training_split,
+            solution_method=self.config.solution_method,
+            multinomials=self.config.multinomials,
+            number_of_crossvalidations=self.config.number_of_crossvalidations,
+            overwrite=True
+        )
+        variable_headers = model.get_feature_vector()
+        if self.config.extra_features is not None:
+            # create additional terms
+            try:
+                add_terms = self.config.extra_features
+                for j in model.regression_data_columns:
+                    add_terms = [
+                        add_terms[k].replace(j, "variable_headers['" + str(j) + "']")
+                        for k in range(0, len(add_terms))
+                    ]
+                model.set_additional_terms(
+                    [
+                        eval(m, GLOBAL_FUNCS, {"variable_headers": variable_headers})
+                        for m in add_terms
+                    ]
+                )
+            except:
+                raise ValueError("Additional features could not be constructed.")
+        return model
+
+    def _get_metrics(self, model):
+        return {"RMSE": model.errors["MSE"] ** 0.5, "R2": model.errors["R2"]}
+
+
+class PysmoRBFTrainer(PysmoTrainer):
+    """Train a radial basis functional surrogate model."""
+
+    # model_type will be this with the basis function prepended, separated by a space
+    base_model_type = "rbf"
```

Comment:
Is it necessary to have both the `base_model_type` and `model_type` attributes (seeing as they are the same)? I note the other trainers I have seen so far only have the `model_type` attribute.

Reply:
Yes. The treason we have two of them is that we plan eventually for the RBFs to eventually report the exact type of RBF being trained (cubic, gaussian etc) as ``model_type``. In that case, ``base_model_type``  and ``model_type`` would contain different information